### ISSUE TITLE
DDLDatabase scripts path wild cards do not work correctly

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#7767]
When using wild cards in the scripts path property of the `DDLDatabase` configuration, there are a few things that don't work correctly, currently. E.g.:

### Full qualification

This doesn't work because the path isn't fully qualified:

```xml
<value>src/main/resources/db/migration/*.sql</value>
```

This works (in Maven):

```xml
<value>${basedir}/src/main/resources/db/migration/*.sql</value>
```

Obviuosly, paths don't need to be fully qualified.

### Wildcards in directory names

This doesn't work because there is a `file.exists()` check to see if the path stripped of the first wild card exists:

```xml
<value>${basedir}/src/main/resources/db/mig*/*.sql</value>
```

So, there is some logic that checks for the existence of the directory `${basedir}/src/main/resources/db/mig`, which is nonesense of course. We should backtrack to the first directory up the path that *does* exist. E.g. if this doesn't exist:

```
${basedir}/src/main/resources/db/mig
```

Then maybe this does:

```
${basedir}/src/main/resources/db
```

Etc.

----
See also: #7766

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.12. Backport scheduled for 3.11.5: #7768

**Comment by lukaseder** [comment]
This fix worked around a few issues but wasn't really correct. A better fix is on its way via #9726
